betterlockscreen

```diff
@@ -379,16 +380,59 @@ get_wall_list() {
 get_image() {
 
     local path="$1"
+    local display="$2"
 
     # we have a file
     if [ -f "$path" ]; then
         WALL_LIST+=("$path")
         return
     # we have a directory
     elif [ -d "$path" ]; then
-        dir=("$path"/*)
-        rdir="${dir[RANDOM % ${#dir[@]}]}"
-        get_image "$rdir" # <-- calls itself
+        # prefer closest scale
+        if [ "$prefer_closest_resolution" = true ] && [ -n "$display" ]; then
```

Comment:
`--span` doesn't set $display, so we fail this check.

Reply:
This is kind of intentional, I didn't want to have this working for span image. But thinking about it now, I don't see why not, probably just because I'm not using it :)